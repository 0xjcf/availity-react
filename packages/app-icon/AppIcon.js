import React from 'react';
import PropTypes from 'prop-types';

function AppIcon({
  tag: Tag,
  color,
  size,
  branded,
  className,
  children,
  ...props
}) {
  const classname = [
    className,
    'app-icon',
    branded ? `app-icon-branded-${color}` : `app-icon-${color}`,
    size && `app-icon-${size}`,
  ]
    .filter(a => a)
    .join(' ');

  return (
    <Tag {...props} className={classname}>
      {children}
      {branded && <span className="caret" />}
    </Tag>
  );
}

AppIcon.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  branded: PropTypes.bool,
  className: PropTypes.string,
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  children: PropTypes.node,
};

AppIcon.defaultProps = {
  tag: 'span',
  color: 'black',
};

export default AppIcon;
