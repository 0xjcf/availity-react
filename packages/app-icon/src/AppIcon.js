import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function AppIcon({ tag: Tag, color, size, src: image, alt, branded, className, children, ...props }) {
  const classes = classNames(className, 'app-icon', {
    [`app-icon-${color}`]: color && !branded,
    [`app-icon-branded-${color}`]: color && branded,
    [`app-icon-${size}`]: size,
    'border-0': image,
  });

  return (
    <Tag {...props} className={classes}>
      {image ? <img className="w-100 h-100 align-baseline" src={image} alt={alt} /> : children}
      {branded && <span className="caret" />}
    </Tag>
  );
}

AppIcon.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  branded: PropTypes.bool,
  className: PropTypes.string,
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  children: PropTypes.node,
  src: PropTypes.string,
  alt: PropTypes.string,
};

AppIcon.defaultProps = {
  tag: 'span',
  color: 'black',
};

export default AppIcon;
