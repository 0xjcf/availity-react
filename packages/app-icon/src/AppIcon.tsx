import React from 'react';
import classNames from 'classnames';

export type AppIconProps = {
  /** For src prop. the alt property for your image source is not found or is loading. */
  alt?: string;
  /** Triggers "branded" styles. */
  branded?: boolean;
  /**  */
  children?: React.ReactNode;
  /**  */
  className?: string;
  /** Potential values: "black", "blue", "green", "orange", "red". */
  color?: string;
  /** Potential values: "lg", "xl". */
  size?: string;
  /** If image source is provided, it will render this instead of children. */
  src?: string;
  /**  */
  tag?: React.ElementType;
  /**  */
  title?: string;
};

function AppIcon({
  tag: Tag = 'span',
  color = 'black',
  size,
  src: image,
  alt,
  branded,
  className,
  children,
  ...props
}: AppIconProps): JSX.Element {
  const classes = classNames(className, 'app-icon', {
    [`app-icon-${color}`]: color && !branded,
    [`app-icon-branded-${color}`]: color && branded,
    [`app-icon-${size}`]: size,
    'border-0': image,
  });

  return (
    <Tag {...props} className={classes}>
      {image ? <img className="w-100 h-100 align-baseline" src={image} alt={alt} /> : children}
      {branded && <span className="caret" />}
    </Tag>
  );
}

export default AppIcon;
