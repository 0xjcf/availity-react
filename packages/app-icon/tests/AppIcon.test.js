import React from 'react';
import renderer from 'react-test-renderer';
import AppIcon from '..';

describe('AppIcon', () => {
  test('should render', () => {
    const component = renderer.create(<AppIcon />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render color', () => {
    const component = renderer.create(<AppIcon color="green" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render size', () => {
    const component = renderer.create(<AppIcon size="lg" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render children', () => {
    const component = renderer.create(<AppIcon>AI</AppIcon>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render branded', () => {
    const component = renderer.create(<AppIcon branded />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render additional classNames', () => {
    const component = renderer.create(<AppIcon className="and-more" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render additional attributes', () => {
    const component = renderer.create(<AppIcon title="Availity Icon" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render custom tags', () => {
    const component = renderer.create(<AppIcon tag="i" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
