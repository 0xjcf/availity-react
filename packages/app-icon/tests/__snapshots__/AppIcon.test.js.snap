// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon should render 1`] = `
<span
  className="app-icon app-icon-black"
/>
`;

exports[`AppIcon should render additional attributes 1`] = `
<span
  className="app-icon app-icon-black"
  title="Availity Icon"
/>
`;

exports[`AppIcon should render additional classNames 1`] = `
<span
  className="and-more app-icon app-icon-black"
/>
`;

exports[`AppIcon should render branded 1`] = `
<span
  className="app-icon app-icon-branded-black"
>
  <span
    className="caret"
  />
</span>
`;

exports[`AppIcon should render children 1`] = `
<span
  className="app-icon app-icon-black"
>
  AI
</span>
`;

exports[`AppIcon should render color 1`] = `
<span
  className="app-icon app-icon-green"
/>
`;

exports[`AppIcon should render custom tags 1`] = `
<i
  className="app-icon app-icon-black"
/>
`;

exports[`AppIcon should render size 1`] = `
<span
  className="app-icon app-icon-black app-icon-lg"
/>
`;
